kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -1969,6 +1972,50 @@ bool IndexerASTVisitor::VisitInitListExpr(const clang::InitListExpr* ILE) {
   return true;
 }
 
+bool IndexerASTVisitor::TraverseCallExpr(clang::CallExpr* CE) {
+  auto callee = CE->getDirectCallee();
+  auto callee_exp = CE->getCallee();
+  bool valid = callee != nullptr && callee_exp != nullptr &&
+               callee->param_size() == CE->getNumArgs();
+  for (unsigned arg = 0; valid && arg < CE->getNumArgs(); ++arg) {
+    valid = CE->getArg(arg) != nullptr && callee->getParamDecl(arg) != nullptr;
+  }
+  if (valid) {
+    auto callee_node = BuildNodeIdForDecl(callee);
+    if (!WalkUpFromCallExpr(CE)) return false;
+    if (!TraverseStmt(callee_exp)) return false;
+    for (unsigned arg = 0; arg < CE->getNumArgs(); ++arg) {
```

Comment:
for (clang::Expr* arg : CE->arguments()) {
  ...
}

Reply:
Doesn't work with callee->getParamDecl(arg) below.